### PR TITLE
feat(core): remove not allowed roles

### PR DESCRIPTION
**Pull request by sarkapalkovicova**
Depending on where the request comes from, remove roles that are not allowed.

### CODE REVIEW

File: perun-rpc/src/main/java/cz/metacentrum/perun/rpc/Api.java [hunk 1]
```diff
@@ -225,6 +225,8 @@ private static PerunPrincipal setupPerunPrincipal(HttpServletRequest req, Deseri
 		int extSourceLoa;
 		Map<String, String> additionalInformations = new HashMap<>();
 
+		String referer = req.getHeader("Referer");
```
Comment (zlamalp) [inline perun-rpc/src/main/java/cz/metacentrum/perun/rpc/Api.java:228]: I'm not really sure what this logic actually suppose to do, but HTTP request headers are set by client. Hence I can forge a request using cURL with specific `Referer` value, even it if originates from my script. If this suppose to somehow override the problems with MFA, I would be cautious.

Comment (sarkapalkovicova) [reply perun-rpc/src/main/java/cz/metacentrum/perun/rpc/Api.java:228]: According to the referer header, this code removes roles from a session that are not defined as allowed in the config. It does not skip MFA but only limits user's rights. So even if a user fakes the request, there should be no security risk.

**Review by bodnara — COMMENTED** [review]
Should this PR also contain development note about the new config property or is it not necessary.

File: perun-base/src/main/resources/perun-base.xml [hunk 2]
```diff
@@ -82,6 +82,7 @@
 		<property name="mfaAuthTimeout" value="${perun.introspectionEndpoint.mfaAuthTimeout}"/>
 		<property name="mfaAuthTimeoutPercentageForceLogIn" value="${perun.introspectionEndpoint.mfaAuthTimeoutPercentageForceLogIn}"/>
 		<property name="enforceMfa" value="${perun.enforceMfa}"/>
+		<property name="appAllowedRoles" value="#{'${perun.appAllowedRoles.apps}'.split('\s*,\s*')}"/>
```
Comment (xflord) [inline perun-base/src/main/resources/perun-base.xml:85]: Looks like a good solution, just please include deployment notes related to the new header and new config property (an explanation of its structure would be nice)

**Review by bodnara — APPROVED** [review]

**Review by xflord — APPROVED** [review]

**Pull request merged**